### DISCUSSION TITLE
Option to generate a subscription feed locally

### DISCUSSION THREAD

**Discussion by matt-from-wii-sports-2** [libre-tube/LibreTube#6656] [Ideas]
### Describe your suggested feature

This feature request defies the purpose of this app. But since there is an option for the local video data extraction, and Piped has problems with generating an up to date feed, this could be very beneficial at least for some use cases

### Other details

- Will fix #6548
- Probably should be a separate toggle in the settings
- Should show a progress bar with the ability to tap a button to stop loading
- NewPipe and Grayjay load subscriptions like this

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Akash-Sriram** [comment]
Instead of using "piped to load," it should use "direct newpipe extractor" to achieve this, I guess.

**Comment by Akash-Sriram** [comment]
Its would be greate if we directly use newpipe extract to get subscription feed and search results instead of using pipped servers in between this will fix the issue we could inform the user with caution and its user's preference.

**Comment by voidplayer** [comment]
I like to use libre tube as is much nicer interface, but as you said, the list of videos is always missing 90% of them, so i have to run an update feeds on a local new pipe (it takes almost a minute to update my feed list in) while a watch the few videos that load on libretube feed

I should separate them into 2 feeds, channels that update videos daily and channels that rarely update to make the feed update faster

**Comment by Akash-Sriram** [comment]
According to the roadmap, **direct scraping without Piped (e.g., by using NewPipe Extractor directly)** is explicitly **not planned**, which is fine by me. However, my use case slightly differs: I synchronize both LibreTube and Pipe.Videos using the same subscription list and instance. In Pipe.Videos, around 90% of my feeds appear correctly, with only occasional or very recent videos missing, which is manageable. In contrast, LibreTube displays significantly more inconsistencies. This raises the question: what could be causing this difference?

Below is a side-by-side comparison of the feeds:

<div style="display: flex; align-items: center; justify-content: center;">
  <img src="https://github.com/user-attachments/assets/52226079-ba39-4889-95ec-a05c92ff9f91" alt="Pipe.Videos Feed" width="40%" height="auto" style="margin-right: 10px;" />
  <img src="https://github.com/user-attachments/assets/43186051-190e-433f-b588-0a6b3c97c073" alt="LibreTube Feed" width="40%" height="auto" />
</div>

**Comment by Bnyro** [comment]
Local feed extraction added in https://github.com/libre-tube/LibreTube/pull/6938
